units/patch_intent.go

```diff
@@ -339,7 +335,9 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 		}
 	}
 
-	if len(patchDoc.VariantsTasks) == 0 {
+	skipForFailed := failedOnly && !(previousPatchStatus == evergreen.PatchFailed)
+
+	if len(patchDoc.VariantsTasks) == 0 && !skipForFailed {
```

Comment:
Why do you want to skip this for failedOnly? I think there's actually a good argument to build pairs for failed, to ensure dependencies are added (since dependencies theoretically were successful and not failed).

Reply:
this exits if it's  using the failedOnly flag but the previous task didn't fail, meaning there's nothing to do since there are no tasks to add because nothing failed.